frontend/src/app/shared/components/show-changes/show-changes.component.ts

```diff
@@ -0,0 +1,251 @@
+import { IDay } from 'core-app/core/state/days/day.model';
+import { take } from 'rxjs/operators';
+import { TimezoneService } from 'core-app/core/datetime/timezone.service';
+import { ConfigurationService } from 'core-app/core/config/configuration.service';
+
+@Component({
+  selector: 'op-show-changes',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './show-changes.component.html',
+  styleUrls: ['./show-changes.component.sass'],
+})
+export class OpShowChangesComponent extends UntilDestroyedMixin implements AfterViewInit {
+  @HostBinding('class.op-show-changes') className = true;
+
+  public opened = false;
+
+  public dropDownDescription = '';
+
+  public nonWorkingDays:IDay[] = [];
+
+  public selectedDate = '';
+
+  public selectedTimezoneFormattedTime = '';
+
+  public filterSelected = false;
+
+  public timeZoneSelected = false;
+
+  public daysNumber = 0;
+
+  public tooltipPosition = SpotDropAlignmentOption.BottomRight;
+
+  public text = {
+    toggle_title: this.I18n.t('js.show_changes.toggle_title'),
+    header_description: this.I18n.t('js.show_changes.header_description'),
+    clear: this.I18n.t('js.show_changes.clear'),
+    apply: this.I18n.t('js.show_changes.apply'),
+    show_changes_since: this.I18n.t('js.show_changes.show_changes_since'),
+    time: this.I18n.t('js.show_changes.time'),
+    help_description: this.I18n.t('js.show_changes.help_description'),
+    timeZone: this.configuration.isTimezoneSet() ? moment().tz(this.configuration.timezone()).zoneAbbr() : 'local',
+    time_description: () => this.I18n.t('js.show_changes.time_description', { time: this.selectedTimezoneFormattedTime, days: this.daysNumber }),
+  };
+
+  public showChangesAvailableValues = [
+    {
+      value: '0',
+      title: this.I18n.t('js.show_changes.drop_down.none'),
+    },
+    {
+      value: '1',
+      title: this.I18n.t('js.show_changes.drop_down.yesterday'),
+    },
+    {
+      value: '2',
+      title: this.I18n.t('js.show_changes.drop_down.last_working_day'),
+    },
+    {
+      value: '3',
+      title: this.I18n.t('js.show_changes.drop_down.last_week'),
+    },
+    {
+      value: '4',
+      title: this.I18n.t('js.show_changes.drop_down.last_month'),
+    },
+    {
+      value: '5',
+      title: this.I18n.t('js.show_changes.drop_down.a_specific_date'),
+    },
+    {
+      value: '6',
+      title: this.I18n.t('js.show_changes.drop_down.between_two_specific_dates'),
+    },
+  ];
+
+  public query$ = this.wpTableFilters.querySpace.query.values$();
+
+  constructor(
+    readonly I18n:I18nService,
+    readonly wpTableFilters:WorkPackageViewFiltersService,
+    readonly halResourceService:HalResourceService,
+    private weekdaysService:WeekdayService,
+    private daysService:DayResourceService,
+    readonly timezoneService:TimezoneService,
+    private configuration:ConfigurationService,
+  ) {
+    super();
+  }
+
+  async ngAfterViewInit():Promise<void> {
```

Comment:
Angular lifecycle hoooks do not wait for whatever is in the ngOnInit/ngAfterViewInit hooks. If you need to wait for it, turn it into a promise/observable and wait for it in the template with `ngIf="nonWorkingDays$ | async; else loading`.